addons/binding/org.openhab.binding.helios/ESH-INF/thing/channel-types.xml

```diff
@@ -0,0 +1,173 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<thing:thing-descriptions bindingId="helios" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xmlns:thing="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0"
+    xsi:schemaLocation="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0 http://eclipse.org/smarthome/schemas/thing-description-1.0.0.xsd">
+
+    <!-- Helios Channel Types -->
+
+    <channel-type id="keypressedstamp">
```

Comment:
If I see it correctly from the code, all those "timestamp" channels are NOT a feature of the hardware, but you simply derive them yourself. This is imho an anti-pattern as you could potentially do this for ANY channel in ANY binding. If everybody would do this, we would end up with a doubled amount of channels. I hope you agree that this is not a desired outcome.
I'd therefore suggest to remove those channels here and jointly look for a better solution in https://github.com/eclipse/smarthome/issues/4748 (while you can simply create simply rules as a workaround for your own setup meanwhile).

Reply:
> If I see it correctly from the code, all those "timestamp" channels are NOT a feature of the hardware, but you simply derive them yourself

Ah.. I do not in fact. See https://github.com/openhab/openhab2-addons/pull/3041/files#diff-1d2980eb11ca700f7e42b1e475aab4dcR700. The remote devices do stamp the events that occur

Reply:
Ah, ok. Is this information really helpful in all those cases? I mean, isn't the timestamp more or less the same time that ESH gets the event and thus doesn't have much additional meaning? Or can you receive events, which have a timestamp from e.g. 30 minutes ago?

Reply:
@kaikreuzer Yes indeed. The Helios uses a subscription-based model in which you poll the device and instruct it to send all internal logs since a given point in time. You can thus determine yourself at which interval you requests any new logs. The Helios as a complicated automation engine onboard which allows you "program" it do do certain stuff, such as opening doors, capturing video, block users from entering codes and so forth. In order to keep things a bit in sync with on-board logic, and stuff that is running in another system (e.g. ESH), the timestamps are essential. (e.g. you can send commands to the Helios by various means, including the REST interface it exposes). It is this better not to throw away these stamps.

Reply:
Ok, thank you for the detailed clarification, let's keep them as they are!